src/main/java/org/apache/sling/scripting/jsp/taglib/helpers/XSSSupport.java

```diff
@@ -16,56 +16,89 @@
  */
 package org.apache.sling.scripting.jsp.taglib.helpers;
 
-import org.owasp.esapi.ESAPI;
+import org.apache.sling.xss.XSSAPI;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
 
 /**
  * Support for basic XSS protection as provided by the OWASP ESAPI's escape
  * methods.
  */
+@Component
 public class XSSSupport {
-
-    /**
-     * The encoding modes supported by this tag.
-     */
-    public enum ENCODING_MODE {
-        /**
-         * Encodes the content as HTML
-         */
-        HTML, HTML_ATTR, JS, XML, XML_ATTR
+    
+    @Reference
+    private XSSAPI xssApi;
+    
+    private static XSSAPI XSS_API;
```

Comment:
I believe this field needs to be volatile.

Reply:
Good catch, thank you.